api/v1/nodescan/service.go

```diff
@@ -95,32 +96,38 @@ func featureVersionToKernelComponent(fv database.FeatureVersion) *v1.GetNodeVuln
 
 func (s *serviceImpl) evaluateLinuxKernelVulns(req *v1.GetNodeVulnerabilitiesRequest) (string, []*v1.Vulnerability, *v1.GetNodeVulnerabilitiesResponse_KernelComponent, error) {
 	osImage := strings.ToLower(req.GetOsImage())
+	kernelVersion := req.GetKernelVersion()
 
 	var match *kernelparser.ParseMatch
+Parsers:
 	for name, parser := range kernelparser.Parsers {
-		var ok bool
 		var err error
-		match, ok, err = parser(s.db, req.GetKernelVersion(), osImage)
-		if err != nil {
-			return "", nil, nil, err
-		}
-		if !ok {
-			continue
-		}
-		if match == nil {
-			log.Debugf("%s %s matched %s, but no match found", osImage, req.GetKernelVersion(), name)
-			return "", nil, nil, nil
+		match, err = parser(s.db, kernelVersion, osImage)
+		switch err {
+		case nil:
+			// Found a match.
+			break Parsers
+		case kernelparser.ErrKernelUnrecognized:
+			// This parser did not recognize the kernel. Try another one.
+			continue Parsers
+		case kernelparser.ErrKernelUnsupported:
+			log.Debugf("%s parser found unsupported kernel: %s %s", name, req.GetOsImage(), kernelVersion)
+		case kernelparser.ErrNodeUnsupported:
+			log.Debugf("%s parser found unsupported node: %s %s", name, req.GetOsImage(), kernelVersion)
+		default:
+			// Should never happen.
+			utils.Must(errors.Errorf("Found unexpected error in kernel parser: %v", err))
 		}
```

Comment:
Wouldn't it make more sense to use a Status Enum rather than error codes? It would give us a compile-time assurance instead of a runtime failure. It would also make the function interface less ambiguous, and prevent inadvertent panicking.